src/Data/List/Membership/Propositional/Properties.agda

```diff
@@ -179,6 +179,44 @@ module _ {p} {P : A → Set p} (P? : Decidable P) where
   ∈-filter⁻ : ∀ {v xs} → v ∈ filter P? xs → v ∈ xs × P v
   ∈-filter⁻ = Membershipₛ.∈-filter⁻ (P.setoid A) P? (P.subst P)
 
+------------------------------------------------------------------------
+-- nub and dedup
+
+module _ {ℓ} {R : Rel A ℓ} (R? : B.Decidable R) where
+  ∈-nub-filter⁻ : ∀ y {xs z} → z ∈ nub-filter R? y xs → z ∈ xs
+  ∈-nub-filter⁻ y {x ∷ xs} z∈nub-filter[R?,y,xs] with does (R? y x)
+  ∈-nub-filter⁻ y {x ∷ xs} z∈nub-filter[R?,y,xs]         | true  = there (∈-nub-filter⁻ y z∈nub-filter[R?,y,xs])
+  ∈-nub-filter⁻ y {x ∷ xs} (here refl)                   | false = here P.refl
+  ∈-nub-filter⁻ y {x ∷ xs} (there z∈nub-filter[R?,x,xs]) | false = there (∈-nub-filter⁻ x z∈nub-filter[R?,x,xs])
+
+  ∈-nub⁻ : ∀ xs {z} → z ∈ nub R? xs → z ∈ xs
+  ∈-nub⁻ (x ∷ xs) (here refl) = here refl
+  ∈-nub⁻ (x ∷ xs) (there z∈nub[R?,xs]) = there (∈-nub-filter⁻ x z∈nub[R?,xs])
+
+  ∈-dedup⁻ : ∀ xs {z} → z ∈ dedup R? xs → z ∈ xs
+  ∈-dedup⁻ (x ∷ xs) (here refl) = here refl
+  ∈-dedup⁻ (x ∷ xs) (there z∈dedup[R,xs]) = there (∈-dedup⁻ xs (proj₁ (∈-filter⁻ (¬? ∘ R? x) z∈dedup[R,xs])))
+
+module _ {R : Rel A ℓ} (_≈?_ : B.Decidable (_≡_ {A = A})) where
```

Comment:
It's neater to write this as `B.Decidable {A = A} _≡_`. We also tend to name this as `_≟_`.

Reply:
Also what is the extra `{R : Rel A ℓ}` for?

Reply:
Finally shouldn't these proofs belong in `Membership.DecPropositional.Properties` if we have decidable equality?